frontend/src/components/navigator/PrivateRoute.jsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import {useAuth} from '../../contexts/AuthContext';
+
+const PrivateRoute = () => {
+  // {component: Component, ...rest}
+  const { currentUser } = useAuth();
+  const location = useLocation();
+
+  return currentUser ? <Outlet /> : <Navigate to="/login" state={{ from: location.pathname }}/>;
+};
```

Comment:
What is Outlet?

Reply:
A placeholder for rendering the content of nested routes.